onnxruntime/test/providers/kernel_compute_test_utils.cc

```diff
@@ -0,0 +1,171 @@
+    auto cuda_ep = DefaultCudaExecutionProvider();
+    ep_type = cuda_ep->Type();
+    auto cuda_transfer = cuda_ep->GetDataTransfer();
+    ASSERT_STATUS_OK(dtm.RegisterDataTransfer(std::move(cuda_transfer)));
+    ASSERT_STATUS_OK(execution_providers.Add(ep_type, std::move(cuda_ep)));
+  }
+#endif
+#ifdef USE_ROCM
+  if (provider_ == kRocmExecutionProvider) {
+    auto rocm_ep = DefaultRocmExecutionProvider();
+    ep_type = rocm_ep->Type();
+    auto rocm_transfer = rocm_ep->GetDataTransfer();
+    ASSERT_STATUS_OK(dtm.RegisterDataTransfer(std::move(rocm_transfer)));
+    ASSERT_STATUS_OK(execution_providers.Add(ep_type, std::move(rocm_ep)));
+  }
+#endif
+
+  Model model("test", false, ModelMetaData(), ORT_TSTR(""), IOnnxRuntimeOpSchemaRegistryList(),
+              {{domain_, opset_version_}}, {}, DefaultLoggingManager().DefaultLogger());
+
+  std::vector<NodeArg*> input_args;
+  std::unordered_map<std::string, OrtValue> initializer_map;
+  for (auto& data : input_data_) {
+    input_args.emplace_back(&data.def_);
+    const auto& name = data.def_.Name();
+    // If running on CPU or input is CPU input, use the OrtValue from input_data_ directly.
+    // Otherwise, we need to create a new OrtValue on target device.
+    if (provider_ == kCpuExecutionProvider || data.is_cpu_data_) {
+      initializer_map[name] = data.value_;
+    }
+#if defined(USE_CUDA) || defined(USE_ROCM)
+    if ((provider_ == kCudaExecutionProvider || provider_ == kRocmExecutionProvider) && !data.is_cpu_data_) {
+      OrtValue gpu_value;
+      const Tensor& tensor = data.value_.Get<Tensor>();
+      Tensor::InitOrtValue(tensor.DataType(), tensor.Shape(),
+                           execution_providers.Get(ep_type)->GetAllocator(0, OrtMemTypeDefault), gpu_value,
+                           tensor.Strides());
+      ASSERT_STATUS_OK(dtm.CopyTensor(tensor, *gpu_value.GetMutable<Tensor>()));
+      initializer_map[name] = gpu_value;
+    }
+#endif
+  }
+
+  std::vector<NodeArg*> output_args;
+  for (auto& data : output_data_) {
+    output_args.emplace_back(&data.def_);
+  }
+
+  Graph& graph = model.MainGraph();
+  auto& node = graph.AddNode("node", op_, op_, input_args, output_args, nullptr, domain_);
+  for (auto& add_attribute_fn : add_attribute_funcs_) {
+    add_attribute_fn(node);
+  }
+  ASSERT_STATUS_OK(graph.Resolve());
+
+  node.SetExecutionProviderType(ep_type);
+  OptimizerExecutionFrame::Info info({&node}, initializer_map, graph.ModelPath(), *execution_providers.Get(ep_type),
+                                     [](std::string const&) { return false; });
+  const KernelCreateInfo* kernel_create_info = nullptr;
+  ASSERT_STATUS_OK(info.TryFindKernel(&node, &kernel_create_info));
+  ASSERT_TRUE(kernel_create_info);
+
+  // If the tester specifies output i is strided tensor output, we need to check output i for this kernel is
+  // MayStridedOutput from the KernelDef, and create an OrtValue as output by sharing the data buffer from
```

Comment:
if we handled those here, how our allocation planner got tested to match what you mentioned in the comment ?

Reply:
As mentioned in another comment, it's tested in allocation_planner_test.cc